### PR TITLE
Import everything non-GLL-specific from rust-lang/gll.

### PR DESCRIPTION
**Pull request by eddyb**
Let me know if you spot anything GLL-specific left around.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -14,3 +14,5 @@ description = "Grammar framework."
 
 [dependencies]
 indexmap = "1"
+indexing = "0.3.2"
```
Comment (CAD97) [inline Cargo.toml:17]: Side note: I still very much want to finish the polish for [`windex`](https://github.com/cad97/windex) so maybe we can use that here. Definitely not blocking, just a side thought.

File: src/forest.rs [hunk 2]
```diff
@@ -0,0 +1,563 @@
...
+}
+
+pub type OwnedParseForestAndNode<G, P, I> = ExistsL<PairL<ParseForestL<G, I>, ParseNodeL<P>>>;
+
+#[derive(Debug)]
+pub struct MoreThanOne;
+
+impl<'i, P, G, I: Input> ParseForest<'i, G, I>
+where
+    // FIXME(eddyb) these shouldn't be needed, as they are bounds on
```
Comment (CAD97) [inline src/forest.rs:64]: Is there a rust issue reference? It should probably be named here.

**Review by CAD97 — APPROVED** [review]
Everything looks good and not specific to gll to me!

File: src/forest.rs (hunk 2 above)
Comment (eddyb) [reply src/forest.rs:64]: This is not a bug, but rather a missing feature (definitions implying bounds onto uses).

**Pull request merged**